src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Deposit.tsx

```diff
@@ -204,40 +202,18 @@ export const Deposit: React.FC<DepositProps> = ({
     enableEstimateFees: Boolean(!isApprovalRequired && bnOrZero(inputValues?.cryptoAmount).gt(0)),
   })
 
-  useEffect(() => {
-    if (!accountId) return
-    ;(async () => {
-      const isApprovalRequired = await (async () => {
-        // Router contract address is only set in case we're depositting a token, not a native asset
-        if (!inboundAddress) return false
-        // Do not try to get allowance for native assets, including non-EVM ones.
-        if (!isTokenDeposit) return false
-
-        const allowanceOnChainCryptoBaseUnit = await getErc20Allowance({
-          address: fromAssetId(assetId).assetReference,
-          spender: inboundAddress,
-          from: fromAccountId(accountId).account,
-          chainId: asset.chainId,
-        })
-
-        const cryptoAmountBaseUnit = toBaseUnit(inputValues?.cryptoAmount, asset.precision)
-
-        if (bnOrZero(allowanceOnChainCryptoBaseUnit).eq(0)) return true
-        if (bn(cryptoAmountBaseUnit).gt(allowanceOnChainCryptoBaseUnit)) return true
+  const { isAllowanceResetRequired, isApprovalRequired: _isApprovalRequired } =
+    useIsApprovalRequired({
+      assetId,
+      // TODO(gomes): consolidate
```

Comment:
q: what do you mean? removing those optional chaining?

Reply:
monkey see monkey durr 8836993170f60df80df6229b25023fcda5758d13